qutebrowser/browser/commands.py

```diff
@@ -952,7 +953,14 @@ def tab_select(self, index=None, count=None):
         if count is not None:
             index = str(count)
 
-        tabbed_browser, tab = self._resolve_tab_index(index)
+        tabbed_browser, tab = None, None
+        try:
+            tabbed_browser, tab = self._resolve_tab_index(index)
+        except cmdutils.CommandError:
```

Comment:
Silently switching to opening a tab on *any* error seems weird - there are various different errors in `_resolve_tab_index`, and this would just obscure all of them, and make it really hard to find out what's wrong.

Can you talk a bit more about your use-case for this? I'm assuming that you're using `:tab-select --open` with a URL in this case or something? I suppose we could make the "No matching tab for: ..." a special exception class, and then only handle that?

Reply:
Actually, I've looked through `_resolve_tab_index`, and it raises `cmdutils.CommandError` three more times, but those points are reached only if the index is numeric like `[win_id/]tab_id`. The optional arg I propose is intended to be used with a  substring, not index as such, but probably a new exception class might be good anyway (and probably the arg should be renamed from `--open` to e.g. `--new` to avoid possible shadowin of the built-in `open`).

My use-case is that I often need to open specific pages, mostly for local services like `localhost:port` (e.g., interfaces for `Syncthing`, `Transmission` and alike). And I found it convenient to make OS-wide shortcuts to bring me directly to those tabs, even if QB is not a focused window. The only thing that bugged me is that if those tabs weren't open, nothing happened. With the proposed argument, they will just open in this case.

Reply:
And yes, I've read the PRs and issues you linked. They are somewhat related, but their major focus is to modify/extend the functionality of `:open`, which is less non-trivial (and also I am happy with `:open`, I would be suprised when explicetly asking to open a page, it would switch me to an existing instance).

Reply:
Is it Ok if I define the new exception class in the same file `commands.py`? Or are there guidleines for such cases? (didn't find in the info for contributors).